core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/interceptor/ResolveIdentityInterceptor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.interceptor;
+
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.SelectedAuthScheme;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.core.interceptor.SdkInternalExecutionAttribute;
+import software.amazon.awssdk.http.auth.spi.AuthSchemeOption;
+import software.amazon.awssdk.identity.spi.Identity;
+import software.amazon.awssdk.identity.spi.IdentityProvider;
+import software.amazon.awssdk.identity.spi.ResolveIdentityRequest;
+
+@SdkInternalApi
+public class ResolveIdentityInterceptor implements ExecutionInterceptor {
```

Comment:
Since this is not service-specific, is there a reason to use an interceptor for this, or can we just add it to the main execution pipeline(s)?

Reply:
We need this order: auth scheme resolution -> identity resolution -> endpoint resolution. Currently, auth scheme resolution and endpoint resolution are interceptors (and are service specific) that are run in AwsExecutionContextBuilder before the request pipeline, so this felt the most natural. Anything else seems like it would need big refactoring?